version: '1.0'
services:
  app:
    build: .
    command: python3 main.py
    environment:
      POSTGRES_PASSWORD: 12345678
      POSTGRES_MAX_CONNECTIONS: 200
      PG_USER: postgres
      POSTGRES_DB: postgres
      POSTGRES_HOST: db
      POSTGRES_PORT: 5432
      TEMP_DIR: ./bica_tempdata
      LOG_LEVEL: INFO
      BOT_TOKEN: $BOT_TOKEN
    volumes:
      - .:/app
    ports:
      - "5000:5000"
    depends_on:
      db:
        condition: service_healthy
    networks: ['bica-stones-net']
  db:
    image: postgres:17.0
    container_name: bica-stones-db
    environment:
      PGUSER: postgres
      POSTGRES_PASSWORD: 123456789
      POSTGRES_MAX_CONNECTIONS: 200
      POSTGRES_DB: postgres
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./database/creating_tables/creating_tables.sql:/docker-entrypoint-initdb.d/creating_tables.sql
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready", "-U", "postgres", "-d", "postgres", ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks: ['bica-stones-net']
volumes:
  postgres_data:
    driver: local
networks: {bica-stones-net: {}}
